server/pkg/weather/weather_test.go

```diff
@@ -0,0 +1,27 @@
+package weather
+
+import (
+	"testing"
+	"time"
+)
+
+func TestDayOrNight(t *testing.T) {
+	rise, _ := time.Parse("Jan 2, 2006 at 3:04pm (MST)", "Jan 2, 2006 at 6:00am (MST)")
+	set, _ := time.Parse("Jan 2, 2006 at 3:04pm (MST)", "Jan 2, 2006 at 6:00pm (MST)")
+	dayTime, _ := time.Parse("Jan 2, 2006 at 3:04pm (MST)", "Jan 2, 2006 at 12:00pm (MST)")
+	nightTime, _ := time.Parse("Jan 2, 2006 at 3:04pm (MST)", "Jan 2, 2006 at 11:00pm (MST)")
```

Comment:
We should just create the `time.Time` struct directly instead of relying on Parse.

Also, this could be a table test:
```
sunRise := time.Date{2006, 1, 2, 6, 0, 0)  // Jan 2, 2006 at 6:00am
sunSet := time.Date{2006, 1, 2, 18, 0, 0)  // Jan 2, 2006 at 6:00pm

tests := []struct {
		name  string
		current time.Time
		sunRise time.Time
		sunSet time.Time
		want  string
	}{
		{
			name: "testing daytime",
			current: time.Date{2006, 1, 2, 12, 0, 0),
			sunRise: sunRise,
			sunSet: sunSet,
			want: "day",
		}, {
			name: "testing nighttime",
			current: time.Date{2006, 1, 2, 23, 0, 0),
			sunRise: sunRise,
			sunSet: sunSet,
			want: "night",
		},
}
for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got := DayOrNight(tt.current, tt.sunRise, tt.sunSet)
			if got != tt.want {
				t.Errorf("%s got %s want %s", tt.name, got, tt.want)
			}
		})
	}

Reply:
This allows us to add more tests by simply adding more structs to the `tests` variable.